src/treetensornetworks/solvers/alternating_update.jl

```diff
@@ -27,39 +24,41 @@ function _extend_sweeps_param(param, nsweeps)
   return eparam
 end
 
-function process_sweeps(; kwargs...)
-  nsweeps = get(kwargs, :nsweeps, 1)
-  maxdim = get(kwargs, :maxdim, fill(typemax(Int), nsweeps))
-  mindim = get(kwargs, :mindim, fill(1, nsweeps))
-  cutoff = get(kwargs, :cutoff, fill(1E-16, nsweeps))
-  noise = get(kwargs, :noise, fill(0.0, nsweeps))
-
+function process_sweeps(;
+  nsweeps=1,
+  cutoff=fill(1E-16, nsweeps),
+  maxdim=fill(typemax(Int), nsweeps),
+  mindim=fill(1, nsweeps),
+  noise=fill(0.0, nsweeps),
+  kwargs...,
+)
   maxdim = _extend_sweeps_param(maxdim, nsweeps)
   mindim = _extend_sweeps_param(mindim, nsweeps)
   cutoff = _extend_sweeps_param(cutoff, nsweeps)
   noise = _extend_sweeps_param(noise, nsweeps)
-
   return (; maxdim, mindim, cutoff, noise)
 end
 
-function alternating_update(solver, PH, t::Number, psi0::AbstractTTN; kwargs...)
-  reverse_step = get(kwargs, :reverse_step, true)
-
-  nsweeps = _compute_nsweeps(t; kwargs...)
+function alternating_update(
+  solver,
+  PH,
+  psi0::AbstractTTN;
+  checkdone=nothing,
+  order=2,
+  outputlevel=0,
+  time::Number=Inf,
+  time_start=0.0,
+  time_step=time,
+  nsweeps=nothing,
```

Comment:
Since `time`, `time_step`, `time_start`, etc. are specific to the `tdvp` backend, maybe we should only define them in that function? We could just default to `nsweeps=1`, or not set a default for `nsweeps` and expect users/particular solvers to set it.

Reply:
That would move the `_compute_nsweeps` logic into `tdvp`, which to me makes sense since it only seems to be needed by solver backends that actually perform some sort of time evolution.

Reply:
I guess that leaves a question about how to properly track the current time of the evolution, but maybe that can be handled by an `Observer` or by the `solver` that gets passed to `alternating_update` by the `tdvp` wrapper.

Reply:
This could be a nice simplication of the code! I'll look into it & we definitely want to do some version of this redesign.

Reply:
I think at this point we have a good plan for this but it is more involved so should be handled in future PRs.

Reply:
Tracking plans for this in #60.

Reply:
Agreed, yes these definitely need to go but I think the way to do it is the future refactor of the code (next PR after this one) to pass the data structure which carries a named tuple of time step information, and which creates the extra observer function for printing the current time.